### PR TITLE
[Merge]: Dev-mirror to Dev

### PR DESCRIPTION
**Pull request by kalash-copods**

### CODE REVIEW

**Review by arpita-20 — APPROVED** [review]

File: LeaveTrackingApp/serializers/leave_serializers.py [hunk 1]
```diff
@@ -27,6 +29,19 @@ class Meta:
...
+#     class Meta:
+#         model = StatusReason
+#         fields = '__all__'
+
+#     def get_user(self, obj):
+#         return {
+#             'name': obj.user.long_name(),
+#             'profilePicture': obj.user.profile_image
+#         }
+
```
Comment (chandani-copods) [inline LeaveTrackingApp/serializers/leave_serializers.py:44]: If not required, remove this commented code

Comment (kalash-copods) [reply LeaveTrackingApp/serializers/leave_serializers.py:45]: needed for future reference, added TODO comment

File: LeaveTrackingApp/utils.py [hunk 2]
```diff
@@ -170,12 +170,27 @@ def user_leave_stats_user_view(user_id, year_range):
...
+                year_leave_stats['yearly_leaves'].append({
+                    'leaveType': leave_request['leave_type'],
+                    'daysTaken' : len(leave_request['day_details']),
+                    'totalDays': max_days,
+                    'remaining': max(0, max_days - len(leave_request['day_details'])),
+                    'dayDetails': [{'id': day['id'], 'date': day['date']} for day in leave_request['day_details']]
+                })
 
         # Organize quarterly leaves and day details
         leave_wfh_for_year = {
```
Comment (chandani-copods) [inline LeaveTrackingApp/utils.py:196]: look up naming convention .

File: UserApp/serializers/user_serializers.py [hunk 3]
```diff
@@ -4,15 +4,32 @@
...
+        if 'work_type' in data:
+            work_type = data['work_type']
+            work_types = dict(User.WORK_TYPE_CHOICES)
+            if work_type in work_types.values():
+                data['work_type'] = next((k for k, v in work_types.items() if v == work_type), None)
+            else:
+                raise serializers.ValidationError({'work_type': f"Invalid choice: {work_type}. Expected one of {list(work_types.values())}."})
+        return data
+        
+    def to_representation(self, instance):
```
Comment (chandani-copods) [inline UserApp/serializers/user_serializers.py:23]: naming

**Review by chandani-copods — APPROVED** [review]

File: UserApp/serializers/user_serializers.py (hunk 3 above)
Comment (kalash-copods) [reply UserApp/serializers/user_serializers.py:23]: this function is from django serializers. I have overridden it

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: LeaveTrackingApp/models.py [hunk 4]
```diff
@@ -149,6 +158,42 @@ class Leave(models.Model):
...
+    updated_at = models.DateTimeField(auto_now=True)
 
     def __str__(self):
-        return self.leave_type
+        return self.name
+
+class YearPolicy(models.Model):
+    id=models.UUIDField(default=uuid.uuid4, editable=False, unique=True, primary_key=True, verbose_name="Public identifier")
+    year = models.IntegerField()
+    STATUS_CHOICES = [
```
[the remark this answers is not in the document]
Comment (kalash-copods) [reply LeaveTrackingApp/models.py]: resolved

**Review by richa-copods — APPROVED** [review]

**Pull request merged**